server/api_server/tools_routes.py

```
@@ -0,0 +1,685 @@
+            row = cur.fetchone()
+            if row and row['devLastIP']:
+                target = row['devLastIP']
+            else:
+                return jsonify({"error": f"Could not resolve IP for MAC {target}"}), 404
+        finally:
+            conn.close()
+
+    try:
+        # Run nmap -F for fast port scan
+        cmd = ["nmap", "-F", target]
+        result = subprocess.run(
```

Ruff (0.14.6)

`subprocess` call: check for execution of untrusted input

(S603)